src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -575,51 +578,73 @@
                                         (false? (:json_unfolding existing-field))))]
         (remove should-not-unfold? json-fields)))))
 
+(setting/defsetting nested-field-columns-value-length-limit
+  (deferred-tru (str "Maximum length of a JSON string before skipping it during sync for JSON unfolding. If this is set "
+                     "too high it could lead to slow syncs or out of memory errors."))
+  :visibility :internal
+  :export?    true
+  :type       :integer
+  :default    50000)
+
 (defn- sample-json-row-honey-sql
   "Return a honeysql query used to get row sample to describe json columns.
 
   If the table has PKs, try to fetch both first and last rows (see #25744).
   Else fetch the first n rows only."
-  [table-identifier json-field-identifiers pk-identifiers]
+  [driver table-identifier json-field-identifiers pk-identifiers]
   (let [pks-expr         (mapv vector pk-identifiers)
         table-expr       [table-identifier]
-        json-field-exprs (mapv vector json-field-identifiers)]
+        json-field-exprs (mapv (fn [field]
+                                 (if (= (driver.sql/json-field-length driver field) ::driver.sql/nyi)
+                                   [field]
+                                   [[:case
+                                     [:<
+                                      [:inline (nested-field-columns-value-length-limit)]
+                                      (driver.sql/json-field-length driver field)]
+                                     nil
+                                     :else
+                                     field]
+                                    (last (h2x/identifier->components field))]))
+                               json-field-identifiers)]
     (if (seq pk-identifiers)
       {:select json-field-exprs
        :from   [table-expr]
-       ;; mysql doesn't support limit in subquery, so we're using inner join here
```

Comment:
Why remove this comment?

Reply:
whoops

Reply:
put it back in 9b04fb0ca9d8d98a636088162cdbc4eb3f319e78